vcd/resource_vcd_vapp_vm.go

```diff
@@ -812,6 +812,24 @@ func resourceVmHotUpdate(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	if d.HasChange("storage_profile") {
+		var storageProfile types.Reference
+		storageProfilePtr := &storageProfile
+		storageProfileName := d.Get("storage_profile").(string)
+		if storageProfileName != "" {
+			storageProfile, err = vdc.FindStorageProfileReference(storageProfileName)
+			if err != nil {
+				return fmt.Errorf("[vm update] error retrieving storage profile %s : %s", storageProfileName, err)
+			}
+		} else {
```

Comment:
If this `else` clause needed? 
What happens when `UpdateStorageProfile` receives `nil`? Does it reset to something or leave the same (previous) value.
Could we not skip the `UpdateStorageProfile` at all if new value is empty? Probably by checking in some similar fashion as :
```
_,  newStorageProfile := d.GetChange("storage_profile")
if d.HasChange("storage_profile") && newStorageProfile != "" {
    ..      
} 
```

Reply:
change a little differently, but yes that explicit else wasn't needed